warehouse/query-core/src/main/java/datawave/query/jexl/JexlNodeFactory.java

```diff
@@ -108,142 +112,151 @@ public static JexlNode createNodeTreeFromFieldsToValues(ContainerType containerT
                 allValues.addAll(values);
             }
             fieldsToValues.clear();
-            for (String identifier : JexlASTHelper.getIdentifierNames(orgNode)) {
+            for (String identifier : JexlASTHelper.getIdentifierNames(original)) {
                 fieldsToValues.put(identifier, allValues);
             }
         }
         
-        Set<String> fields = fieldsToValues.keySet();
+        List<JexlNode> children = new LinkedList<>();
         
-        JexlNode parentNode = (containerType.equals(ContainerType.OR_NODE) ? new ASTOrNode(ParserTreeConstants.JJTORNODE) : new ASTAndNode(
-                        ParserTreeConstants.JJTANDNODE));
-        int parentNodeChildCount = 0;
+        Set<String> fields = fieldsToValues.keySet();
         
         if (keepOriginalNode) {
-            JexlNodes.ensureCapacity(parentNode, fields.size() + 1);
-            JexlNode child = RebuildingVisitor.copy(orgNode);
-            parentNode.jjtAddChild(child, parentNodeChildCount);
-            child.jjtSetParent(parentNode);
-            parentNodeChildCount++;
+            JexlNode child = copy(original);
+            children.add(child);
             // remove this entry from the fieldsToValues to avoid duplication
-            for (String identifier : JexlASTHelper.getIdentifierNames(orgNode)) {
-                for (Object value : JexlASTHelper.getLiteralValues(orgNode)) {
+            for (String identifier : JexlASTHelper.getIdentifierNames(original)) {
+                for (Object value : JexlASTHelper.getLiteralValues(original)) {
```

Comment:
We can retrieve the values just once instead of for each identifier, like below:
```java
Set<String> identifiers = JexlASTHelper.getIdentifierNames(original);
for (Object value : JexlASTHelper.getLiteralValues(original)) {
    identifiers.forEach((id) -> fieldsToValues.remove(id, value));
}
```

Reply:
Nifty, I like it